src/coreclr/jit/flowgraph.cpp

```diff
@@ -6787,6 +6894,398 @@ void FlowGraphDominanceFrontiers::Dump()
+    // Running this loop in post-order means we end up with sibling links in
+    // reverse post-order. Skip the root since it has no siblings.
+    //
+    for (unsigned i = 0; i < count - 1; i++)
+    {
+        BasicBlock* block  = postOrder[i];
+        BasicBlock* parent = block->bbIPDom;
+        assert(parent != nullptr);
+        assert(reverseDfsTree->Contains(block) && reverseDfsTree->Contains(parent));
+
+        domTree[i].nextSibling                            = domTree[parent->bbReversePostorderNum].firstChild;
+        domTree[parent->bbReversePostorderNum].firstChild = block;
+    }
+
+#ifdef DEBUG
+    if (comp->verbose)
+    {
+        printf("After computing the postdominance tree:\n");
+        for (unsigned i = count; i > 0; i--)
+        {
+            unsigned poNum = i - 1;
+            if (domTree[poNum].firstChild == nullptr)
+            {
+                continue;
+            }
+
+            printf(FMT_BB " :", postOrder[poNum]->bbNum);
+            for (BasicBlock* child = domTree[poNum].firstChild; child != nullptr;
+                 child             = domTree[child->bbReversePostorderNum].nextSibling)
+            {
+                printf(" " FMT_BB, child->bbNum);
+            }
+            printf("\n");
+        }
+        printf("\n");
+    }
+#endif
+
+    // Assign preorder/postorder nums for fast "domnates" queries.
+    class NumberDomTreeVisitor : public DomTreeVisitor<NumberDomTreeVisitor, /* postDom */ true>
+    {
+        unsigned* m_preorderNums;
+        unsigned* m_postorderNums;
+        unsigned  m_preNum  = 0;
+        unsigned  m_postNum = 0;
+
+    public:
+        NumberDomTreeVisitor(Compiler* comp, unsigned* preorderNums, unsigned* postorderNums)
+            : DomTreeVisitor(comp)
+            , m_preorderNums(preorderNums)
+            , m_postorderNums(postorderNums)
+        {
+        }
+
+        void PreOrderVisit(BasicBlock* block)
+        {
+            m_preorderNums[block->bbPostorderNum] = m_preNum++;
+        }
+
+        void PostOrderVisit(BasicBlock* block)
+        {
+            m_postorderNums[block->bbPostorderNum] = m_postNum++;
+        }
+    };
+
+    unsigned* preorderNums  = new (comp, CMK_DominatorMemory) unsigned[count];
+    unsigned* postorderNums = new (comp, CMK_DominatorMemory) unsigned[count];
+
+    NumberDomTreeVisitor number(comp, preorderNums, postorderNums);
+    number.WalkTree(domTree);
+
+    return new (comp, CMK_DominatorMemory)
+        FlowGraphPostDominatorTree(reverseDfsTree, domTree, preorderNums, postorderNums);
+}
+
+//------------------------------------------------------------------------
+// fgComputePostDominanceFrontiers: Compute flow graph dominance frontiers
+//
+// Notes:
+//   Assumes m_postDomTree (and hence m_reverseDfsTree) exist.
+//
+//   Based on "A Simple, Fast Dominance Algorithm" by
+//   Cooper, Harvey & Kennedy
+//
+void Compiler::fgComputePostDominanceFrontiers()
```

Comment:
Do/will we need this? If yes I think it should be abstracted similarly to `FlowGraphDominanceFrontiers`.